src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.Windows.cs

```diff
@@ -17,6 +17,79 @@ public partial class Socket
 
         internal void ReplaceHandleIfNecessaryAfterFailedConnect() { /* nop on Windows */ }
 
+        public Socket(SocketInformation socketInformation)
+        {
+            if (NetEventSource.IsEnabled) NetEventSource.Enter(this);
+
+            InitializeSockets();
+
+            SocketError errorCode = SocketPal.CreateSocket(socketInformation, out _handle,
+                ref _addressFamily, ref _socketType, ref _protocolType);
+
+            if (errorCode != SocketError.Success)
+            {
+                Debug.Assert(_handle.IsInvalid);
+
+                if (errorCode == SocketError.InvalidArgument)
+                {
+                    throw new ArgumentException(SR.net_sockets_invalid_socketinformation, nameof(socketInformation));
+                }
+
+                // Failed to create the socket, throw.
+                throw new SocketException((int)errorCode);
+            }
+
+            if (_handle.IsInvalid)
+            {
```

Comment:
We should dispose of state here that may have been created, namely the _handle.  Even if it's invalid, it'll likely still trigger finalization.

Reply:
This looks like an unnecessary check, since `SocketPal.CreateSocket` already validates handle. Added the disposal there, getting rid of this branch here.